src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php

```diff
@@ -468,4 +470,196 @@ public function getSelfReviewIds(int $employeeNumber, ?array $allowedStatuses =
+            ->setParameter('jobTitle', $jobTitleId);
+        return $q->getQuery()->execute();
+    }
+
+    /**
+     * @param array $reviewerRatings
+     * @return void
+     */
+    public function saveAndUpdateReviewerRatings(array $reviewerRatings): void
+    {
+        $q = $this->createQueryBuilder(ReviewerRating::class, 'reviewerRating');
+
+        foreach (array_values($reviewerRatings) as $i => $reviewerRating) {
+            $reviewerIdParamKey = 'reviewerId_' . $i;
+            $performanceReviewIdParamKey = 'performanceReviewId_' . $i;
+            $kpiIdParamKey = 'kpiId_' . $i;
+
+            /** @var ReviewerRating $reviewerRating */
+
+            $reviewerId = $reviewerRating->getReviewer()->getId();
+            $performanceReviewId = $reviewerRating->getPerformanceReview()->getId();
+            $kpiId = $reviewerRating->getKpi()->getId();
+
+            $q->orWhere(
+                $q->expr()->andX(
+                    $q->expr()->eq('reviewerRating.reviewer', ':' . $reviewerIdParamKey),
+                    $q->expr()->eq('reviewerRating.performanceReview', ':' . $performanceReviewIdParamKey),
+                    $q->expr()->eq('reviewerRating.kpi', ':' . $kpiIdParamKey),
+                )
+            );
+            $q->setParameter($reviewerIdParamKey, $reviewerId)
+                ->setParameter($performanceReviewIdParamKey, $performanceReviewId)
+                ->setParameter($kpiIdParamKey, $kpiId);
+        }
+
+        /** @var array<string, ReviewerRating> $updatableReviewerRatings */
+        $updatableReviewerRatings = [];
+        foreach ($q->getQuery()->execute() as $updatableReviewerRating) {
+            /** @var ReviewerRating $updatableReviewerRating */
+            $itemKey = $this->generateReviewReviewerRatingKey(
+                $updatableReviewerRating->getReviewer()->getId(),
+                $updatableReviewerRating->getPerformanceReview()->getId(),
+                $updatableReviewerRating->getKpi()->getId()
+            );
+            $updatableReviewerRatings[$itemKey] = $updatableReviewerRating;
+        }
+
+        foreach ($reviewerRatings as $key => $reviewerRating) {
+            if (isset($updatableReviewerRatings[$key])) {
+                $updatableReviewerRatings[$key]->setRating($reviewerRating->getRating());
+                $updatableReviewerRatings[$key]->setComment($reviewerRating->getComment());
+
+                $this->getEntityManager()->persist($updatableReviewerRatings[$key]);
+                continue;
+            } else {
+                $this->getEntityManager()->persist($reviewerRating);
+            }
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    /**
+     * @param int $reviewerId
+     * @param int $performanceReviewId
+     * @param int $kpiId
+     * @return string
+     */
+    public function generateReviewReviewerRatingKey(int $reviewerId, int $performanceReviewId, int $kpiId): string
```

Comment:
Doesn't know about the context, but this method seems not interacting with the ORM, is it possible to move this to service layer instead?